src/foam/dao/FileJournal.js

```diff
@@ -171,14 +152,27 @@ foam.CLASS({
           throw new RuntimeException(t);
         }
       `
+    },
+    {
+      class: 'Object',
```

Comment:
Random thought for another PR but could we model a file as a multiton on the filename and use synchronized methods to ensure they share a lock?

Reply:
Not exactly.  The trick here is that we need to use a lock to ensure that complete "records" are written to the file atomically, which can consist of multiple actual "write" calls to the file.  As a journal entry and the comment describing it should be written together.

When/if we model Journal entries then we can remove the synchronization here and only apply at the appropriate level where the entry is actually written to the stream.